### PR TITLE
Demisto SDK Convert Command

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33529

## Description
Creating a new command to support conversion of entities such as Layouts/Classifiers between versions.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by dantavori — CHANGES_REQUESTED** [review]
Great work!

File: demisto_sdk/commands/convert/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+##Note:
+**Classifiers/Mappers conversion from 6.0.0 and above to versions 5.9.9 and below is currently not supported. Request for converting a pack for versions below 6.0.0 will result on layouts conversion only.**
+
+In order to run the command, `DEMISTO_BASE_URL` environment variable should contain the Demisto base URL, and `DEMISTO_API_KEY` environment variable should contain a valid Demisto API Key.
+To set the environment variables, run the following shell commands:
+```
+export DEMISTO_BASE_URL=<YOUR_DESMISTO_BASE_URL>
+export DEMISTO_API_KEY=<YOUR_DEMISTO_API_KEY>
+```
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply demisto_sdk/commands/convert/README.md:13]: We dont need them. I'll delete

File: demisto_sdk/commands/convert/converters/layout/layout_6_0_0_converter.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                if not layout_kind:
+                    continue
+                sections = old_layout.get_layout_sections()
+                tabs = old_layout.get_layout_tabs()
+                if sections:
+                    new_layout_dict[layout_kind] = {'sections': sections}
+                if tabs:
+                    new_layout_dict[layout_kind] = {'tabs': tabs}
+
+            self.update_incident_types_related_to_old_layouts(old_corresponding_layouts, layout_id)
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply demisto_sdk/commands/convert/converters/layout/layout_6_0_0_converter.py:40]: Should be. I'll add it

File: demisto_sdk/commands/convert/converters/layout/layout_up_to_5_9_9_converter.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        Converts new layouts in Layouts dir to the below 6.0.0 layouts convention.
+        Returns:
+            (int): 0 if convert finished successfully, 1 otherwise.
+        """
+        layout_id_to_incident_type = self.layout_to_indicators_or_incidents_dict(self.pack.incident_types)
+        layout_id_to_indicators_dict = self.layout_to_indicators_or_incidents_dict(self.pack.indicator_types)
+        layout_ids_to_convert = [layout for layout in
+                                 self.get_entities_by_entity_type(self.pack.layouts, FileType.LAYOUTS_CONTAINER) if
+                                 layout_id_to_incident_type.get(layout.layout_id()) or layout_id_to_indicators_dict.get(
+                                     layout.layout_id())]
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply demisto_sdk/commands/convert/converters/layout/layout_up_to_5_9_9_converter.py:32]: Nope. This is an edge case. Talked to @ShahafBenYakir it was decided to not support such cases

File: demisto_sdk/commands/convert/converters/classifier/classifier_6_0_0_converter.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        Returns:
+            (None): Creates a new corresponding mapper to 'old_classifier' by 6_0_0 file structure, if mapping exists.
+        """
+        classifier_name_and_id = self.extract_classifier_name(old_classifier)
+        mapping = old_classifier.get('mapping')
+        if not classifier_name_and_id or not mapping:
+            return
+        mapper = dict(id=f'{classifier_name_and_id}-mapper', name=f'{classifier_name_and_id} - Incoming Mapper',
+                      type='mapping-incoming', description='TODO - Add description', version=-1,
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply demisto_sdk/commands/convert/converters/classifier/classifier_6_0_0_converter.py]: After talking to @roysagi , he said the best would be to leave empty description.

File: demisto_sdk/commands/convert/converters/classifier/classifier_6_0_0_converter.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+        Returns:
+            (None): Creates a new corresponding classifier to 'old_classifier' by 6_0_0 file structure.
+        """
+        classifier_name_and_id = self.extract_classifier_name(old_classifier)
+        if not classifier_name_and_id:
+            return
+        new_classifier = {k: v for k, v in old_classifier.to_dict().items() if k in intersection_fields}
+        new_classifier = dict(new_classifier, type='classification', name=f'{classifier_name_and_id} - Classifier',
+                              description='TODO - Add description', fromVersion='6.0.0', id=classifier_name_and_id)
```
[the remark this answers is not in the document]
Comment (tomneeman151293) [reply demisto_sdk/commands/convert/converters/classifier/classifier_6_0_0_converter.py]: After talking to @roysagi , he said the best would be to leave empty description.

File: demisto_sdk/commands/convert/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+## convert
+
+**Convert layouts and classifiers from a new version to an old version and vice versa.**
+
+##Note:
+**Classifiers/Mappers conversion from 6.0.0 and above to versions 5.9.9 and below is currently not supported. Request for converting a pack for versions below 6.0.0 will result on layouts conversion only.**
```
Comment (roysagi) [inline demisto_sdk/commands/convert/README.md:6]: We should really create a simple list of what can be converted, something that is easy to read and gets the info out quickly the first time you approach the command.

**Review by roysagi — CHANGES_REQUESTED** [review]
@tomneeman151293 - Great job this really looks good. I do have some notes:
1) we should try and mitigate the number of test files in the SDK in general, now you are adding about 20 more files - I think you should take another look and use the test suite instead.
2) do not use a prefix separate schema for testing use the real one - if someone breaks the schema and it breaks the command, then the schema change itself should break the unit tests associated with the command.

**Comment by tomneeman151293** [comment]
@roysagi Managed to minimize some test files. There is still work to be done in TestSuite to make it more user friendly, (no funcs to mock classifier/layouts for example, while there is funcs to mock integration etc, which made it a bit difficult for me to mock)

**Review by dantavori — APPROVED** [review]

**Review by roysagi — APPROVED** [review]

**Pull request merged**